src/Cassandra/ISession.cs

```diff
@@ -84,6 +84,12 @@ public interface ISession: IDisposable
         /// <exception cref="InvalidQueryException">When keyspace does not exist</exception>
         void ChangeKeyspace(string keyspaceName);
 
+        /// <summary>
+        /// Switches to the specified keyspace asynchronously.
+        /// </summary>
+        /// <param name="keyspaceName">Case-sensitive name of keyspace to be used.</param>
+        /// <exception cref="InvalidQueryException">When keyspace does not exist</exception>
+        Task ChangeKeyspaceAsync(string keyspaceName);
```

Comment:
Although this change makes sense since we already have a synchronous version of this method, it's still technically a breaking change and I'm worried that there are users out there creating fake classes that implement `ISession` for their tests. In the past there were some methods that were added to interfaces like `IStatement` but it's much less likely that users are creating fakes for these interfaces than `ISession` since this is the main interface of the driver.

Reply:
Responded [here](https://github.com/datastax/csharp-driver/pull/161#issuecomment-473984902)